In [4]:
from classifier import classificeer_vraag, haal_chunks_op

In [1]:
test_vragen = [
        "Hoeveel mobiliteitsbudget krijg ik?",
        "Ik ben mijn kaart kwijt, wat moet ik doen?",
        "Mag ik eerste klas reizen?",
        "Wat is een correctietarief?",
        "Ik ga uit dienst, wat moet ik doen met mijn OV-kaart?",
        "Hoe reserveer ik een deelauto?",
        "Wat zijn de sancties als ik privé rij met de bedrijfswagen?",
        "Hoe werkt de fiscale ruimte van mijn budget?",
        "Wat is de hoofdstad van Nederland?",
        "Ik ben ziek en kan 3 maanden niet reizen, wat moet ik doen?",
    ]

In [5]:
label = classificeer_vraag("Hoeveel mobiliteitsbudget krijg ik?")
print(f"Labels: {label}")

Labels: ['mobiliteitsbudget_overzicht']


In [6]:
chunks = haal_chunks_op(label)

In [7]:
chunks

'Mobiliteitsbudgetten zijn uitsluitend geldig voor medewerkers met wie dit in het arbeidscontract is overeengekomen of aan wie een budget voor een zakelijk auto wordt toegekend. Dit kan gelden voor:\n- Medewerkers die in dienst zijn gekomen na 1-7-2017, een functie hebben in schaal H3 of hoger en de keuze hebben gekregen uit een van de mobiliteitsbudgetten;\n- Medewerkers die in dienst waren vóór 1-7-2017 en die in plaats van een arbeidsvoorwaardelijke leaseauto (categorie B) hebben gekozen voor het OV+budget of het 100% EV-budget;\n- Of voor medewerkers die aantoonbaar >12.500 zakelijke kilometers rijden voor de uitvoering van hun functie.\n\nEr zijn twee budget vormen beschikbaar:\n1. Het OV+budget\n2. Het 100% EV budget\n\nDe keuzeopties zijn vastgelegd in de keuzekaart mobiliteit vanaf functieniveau H3. Jaarlijks wordt deze Keuzekaart geëvalueerd en waar nodig of mogelijk aangepast.\n\nOV+budget:\nHet OV+budget is een OV-Vrij abonnement 2e klasse, plus een maandelijks geldbedrag. E

In [ ]:
for vraag in test_vragen:
        labels = classificeer_vraag(vraag)
        print(f"\nVraag: \"{vraag}\"")
        print(f"Labels: {labels}")

        # Toon chunk-lengtes
        if labels != ["GEEN_MATCH"]:
            chunks = haal_chunks_op(labels)
            print(f"Chunk totaal: {len(chunks)} tekens")

'Mobiliteitsbudgetten zijn uitsluitend geldig voor medewerkers met wie dit in het arbeidscontract is overeengekomen of aan wie een budget voor een zakelijk auto wordt toegekend. Dit kan gelden voor:\n- Medewerkers die in dienst zijn gekomen na 1-7-2017, een functie hebben in schaal H3 of hoger en de keuze hebben gekregen uit een van de mobiliteitsbudgetten;\n- Medewerkers die in dienst waren vóór 1-7-2017 en die in plaats van een arbeidsvoorwaardelijke leaseauto (categorie B) hebben gekozen voor het OV+budget of het 100% EV-budget;\n- Of voor medewerkers die aantoonbaar >12.500 zakelijke kilometers rijden voor de uitvoering van hun functie.\n\nEr zijn twee budget vormen beschikbaar:\n1. Het OV+budget\n2. Het 100% EV budget\n\nDe keuzeopties zijn vastgelegd in de keuzekaart mobiliteit vanaf functieniveau H3. Jaarlijks wordt deze Keuzekaart geëvalueerd en waar nodig of mogelijk aangepast.\n\nOV+budget:\nHet OV+budget is een OV-Vrij abonnement 2e klasse, plus een maandelijks geldbedrag. Een 1e klasse upgrade is mogelijk tegen bijbetaling van €50,- per maand. Met het OV-Vrij abonnement reis je onbeperkt woon-werk, zakelijk en privé met het openbaar vervoer. De hoogte van het maandelijkse bedrag is afhankelijk van je functieniveau en de klasse van het OV-abonnement.\n\nBedragen per functieniveau:\n- E08/E09/TI10/TI11 (Cat. B1): OV+budget €525/mnd, 100% EV-budget €845/mnd\n- E10/H1/H2/TI12/TI13 (Cat. B2/Z): OV+budget €675/mnd, 100% EV-budget €1.035/mnd\n- H3/H4 (Cat. B3/Z3): OV+budget €825/mnd, 100% EV-budget €1.225/mnd\n- H5 en >H5 (Cat. B4): OV+budget €950/mnd, 100% EV-budget €1.375/mnd\n\nJe bent volledig vrij in je besteding van het budget. We gaan er wel vanuit dat je eventuele aanvullende vervoerskosten voor woon-werk en zakelijk verkeer voor eigen rekening neemt. Intercity Direct (toeslag) en incidenteel 1e klasse reizen zijn bij zakelijk verkeer wel toegestaan. Ook is het gebruik van het deelautoconcept toegestaan.\n\n100% EV-budget:\nHet 100% EV-budget is een maandelijks geldbedrag zonder OV-vrij abonnement, met als voorwaarde dat je door middel van een kentekenbewijs of private leasecontract kunt aantonen dat een volledig elektrische auto (100% EV) op jouw naam geregistreerd staat. De medewerker betaalt uit het 100% EV-budget alle reiskosten die in het kader van het dienstverband gemaakt worden. Reis je toch een keer met het OV dan is het declareren van een Intercity Direct (toeslag) en incidenteel 1e klasse reizen zijn bij zakelijk verkeer wel toegestaan.'